components/HttpsComponent/HttpsRequests.c

```diff
@@ -121,41 +87,43 @@ static void https_get_request(esp_tls_cfg_t cfg, const char *WEB_SERVER_URL, con
         }
     } while (written_bytes < strlen(REQUEST));
     ESP_LOGI(TAG, "Reading HTTP response...");
-    do
-    {
+    
+    do {
         len = sizeof(buf) - 1;
         memset(buf, 0x00, sizeof(buf));
-        // vTaskDelay((SEC*5) / portTICK_PERIOD_MS);
+
         ret = esp_tls_conn_read(tls, (char *)buf, len);
 
-        if (ret == ESP_TLS_ERR_SSL_WANT_WRITE || ret == ESP_TLS_ERR_SSL_WANT_READ)
-        {
+        if (ret == ESP_TLS_ERR_SSL_WANT_WRITE || ret == ESP_TLS_ERR_SSL_WANT_READ) {
             continue;
-        }
-        else if (ret < 0)
-        {
-            ESP_LOGE(TAG, "esp_tls_conn_read  returned [-0x%02X](%s)", -ret, esp_err_to_name(ret));
+        } else if (ret < 0) {
+            ESP_LOGE(TAG, "esp_tls_conn_read returned [-0x%02X](%s)", -ret, esp_err_to_name(ret));
             break;
-        }
-        else if (ret == 0)
-        {
+        } else if (ret == 0) {
             ESP_LOGI(TAG, "connection closed");
-
-            vTaskDelete(xTaskHandlerHTTPS);
             break;
         }
+
         len = ret;
-        buf[len + 1] = '\n';
-        xSemaphoreGive(HttpsResponseReadySemaphore);
-        if (xQueueSend(BufQueue1, buf, portMAX_DELAY) != pdTRUE)
-        {
-            ESP_LOGE(TAG, "sending data from Https by queue failed !");
-            vTaskDelete(xTaskHandlerHTTPS);
+        buf[len] = '\0'; // Null-terminate the received data
+
+        
+        if (len > 0) {                  // Check if the read operation completed
+            if (xQueueSend(*HttpsRequestsHandler->ResponseBufQueue, buf, portMAX_DELAY) == pdTRUE)    // Data reading completed; process the data here if needed
+            {
+                ESP_LOGI(TAG, "Reading response data finished, data sent by queue!");
+                xSemaphoreGive(*HttpsRequestsHandler->ResponseReadySemaphore);
```

Comment:
needed mechanism for detecting complete response or broken response and , and if it is broken fix it with something like timeout or waiting for getting complete response ...